### PR TITLE
kubevirt: add ovirt import provider

### PR DESCRIPTION
**Pull request by suomiy**
- share some logic between vmware and ovirt provider
- keep the states and updaters separated for them

few fixes
- better provider cleanup
- fix resource load errors
- allow empty networks for nics in wizard nic modal
- vmware provider fixes + simplification of logic

please see commit messages for more details

### CODE REVIEW

**Comment by suomiy** [comment]
there are still few things missing, but they can be done in a followup

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts [hunk 1]
```diff
@@ -4,20 +4,29 @@ import {
   VMSettingsField,
   RenderableFieldResolver,
   VMWareProviderField,
+  OvirtProviderField,
 } from '../types';
 import { ProvisionSource } from '../../../constants/vm/provision-source';
 
 export const titleResolver: RenderableFieldResolver = {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts:11]: I struggled a bit what should be the strings here? Are we using upstream or downstream name? And it seems a bit too much repeating Red Hat Virtualization everywhere.

@matthewcarleton

File: frontend/packages/kubevirt-plugin/src/components/vm-status/vm-status.tsx [hunk 2]
```diff
@@ -45,17 +45,24 @@ import {
...
 import { getVMLikeModel } from '../../selectors/vm';
 
 import './vm-status.scss';
+import { VMImportKind } from '../../types/vm-import/ovirt/vm-import';
 
 const VIEW_POD_OVERVIEW = 'View pod overview';
 const VIEW_VM_EVENTS = 'View events';
 const IMPORTING_VMWARE_MESSAGE =
   'The virtual machine is being imported from VMware. Disks will be converted to the libvirt format.';
 const IMPORTING_ERROR_VMWARE_MESSAGE = 'The virtual machine could not be imported from VMware.';
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/vm-status/vm-status.tsx:61]: also here

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts (hunk 1 above)
Comment (matthewcarleton) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts:11]: I’ve struggled with that too. I wonder can we get away with in the drop down having Red Hat Virtualization (RHV) and then just use RHV from there on out? We would want to ensure we make it clear to screen readers.

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts:11]: nice, looks much better,

aa

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts:11]: I left the long version int the statuses in the VMs list as there is no repetition

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts:11]: > We would want to ensure we make it clear to screen readers.

Isn't it enough to be the same what normal user sees?  Where would we want to add it? To the label? - which is quite hard as we are using PF `FormGroup` for that

Comment (matthewcarleton) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/renderable-field.ts:11]: ya, you might be right. As long as they make the connection between the provider selection and then the label of the dropdown.

**Comment by suomiy** [comment]
@pcbailey both of these should be fixed already. Maybe you have a wrong commit selected.

**Pull request merged**